### PR TITLE
improved documentation

### PR DESCRIPTION
**Pull request by zippy1981**

### CODE REVIEW

File: src/Microsoft.AspNetCore.Authentication.OpenIdConnect/Events/OpenIdConnectEvents.cs [hunk 1]
```diff
@@ -27,10 +27,12 @@ public class OpenIdConnectEvents : RemoteAuthenticationEvents
         public Func<MessageReceivedContext, Task> OnMessageReceived { get; set; } = context => Task.CompletedTask;
 
         /// <summary>
-        /// Invoked before redirecting to the identity provider to authenticate. This can be used to set ProtocolMessage.State
-        /// that will be persisted through the authentication process. The ProtocolMessage can also be used to add or customize
+        /// Invoked before redirecting to the identity provider to authenticate. This can be used to set
+        /// <see cref="RedirectContext.ProtocolMessage"/>
+        /// that will be persisted through the authentication process. The ProtocolMessage can also be used to add or alter
```
Comment (Tratcher) [inline src/Microsoft.AspNetCore.Authentication.OpenIdConnect/Events/OpenIdConnectEvents.cs]: What were you trying to convey? Or did you just want to add the cref? 

This is confusing two ideas. This sentence was specifically about the ability to round trip user State. Customizing the RedirectContext.ProtocolMessage is already covered in the next sentence.

**Comment by zippy1981** [comment]
Ok I'll have another look tonight and fix.

**Comment by zippy1981** [comment]
@Tratcher ok addressed and added some more docs.

File: src/Microsoft.AspNetCore.Authentication.OpenIdConnect/Events/OpenIdConnectEvents.cs [hunk 2]
```diff
@@ -27,10 +28,17 @@ public class OpenIdConnectEvents : RemoteAuthenticationEvents
...
-        /// Invoked before redirecting to the identity provider to authenticate. This can be used to set ProtocolMessage.State
-        /// that will be persisted through the authentication process. The ProtocolMessage can also be used to add or customize
+        /// Invoked before redirecting to the identity provider to authenticate. This can be used to set
+        /// values in <c>ProtocolMessage.State</c> that will be persisted through the authentication process.
+        /// The <see cref="RedirectContext.ProtocolMessage"/> can also be used to add or alter
         /// parameters sent to the identity provider.
         /// </summary>
+        /// <remarks>
+        /// While this can be used to rewrite the redirect URLs if the app is behind a proxy server, its best to use
+        /// <c>ForwardedHeadersOptions</c>. Please see: https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/proxy-load-balancer?view=aspnetcore-2.1.
```
Comment (Tratcher) [inline src/Microsoft.AspNetCore.Authentication.OpenIdConnect/Events/OpenIdConnectEvents.cs]: We can't use direct links in doc comments, the links are unstable. If it were critical we'd use a forward link.

**Comment by Eilon** [comment]
@Tratcher can you create an aka.ms link for the doc, fix the bad XML, and merge?

**Comment by Tratcher** [comment]
Closing due to lack of follwup. The comments about forwarders do not belong here and the other changes are aesthetic only.

**Pull request closed without merging**